src/bun.js/api/server.zig

```diff
@@ -1560,13 +1560,17 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
             }
         }
 
-        pub fn endWithoutBody(this: *RequestContext, closeConnection: bool) void {
+        /// Complete the response without writing a body payload.
+        /// `sendCrlf` if true will send an \r\n at the end of the HTTP response.
+        /// If we already wrote a body and are using this function to complete the response, then
+        /// `sendCrlf` should be false.
+        pub fn endWithoutBody(this: *RequestContext, closeConnection: bool, sendCrlf: bool) void {
             if (this.resp) |resp| {
                 if (this.flags.is_waiting_for_request_body) {
                     this.flags.is_waiting_for_request_body = false;
                     resp.clearOnData();
                 }
-                resp.endWithoutBody(closeConnection);
+                resp.endWithoutBody(closeConnection, sendCrlf);
```

Comment:
Not for this PR but a good cleanup would be to have a struct so we could have
`resp.endWithoutBody({ .closeConnection = closeConnection, .sendCrlf = sendCrlf })` to avoid the boolean trap.